olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -1018,30 +1018,33 @@ bool DefaultCacheImpl::Release(const DefaultCache::KeyListType& keys) {
     return result;
   }
 
-  for (const auto& key : keys) {
-    if (memory_cache_) {
-      if (!memory_cache_->Remove(key)) {
-        memory_cache_->RemoveKeysWithPrefix(key);
+  if (memory_cache_ || (mutable_cache_ && mutable_cache_lru_)) {
+    for (const auto& key : keys) {
+      if (memory_cache_) {
+        if (!memory_cache_->Remove(key)) {
+          memory_cache_->RemoveKeysWithPrefix(key);
+        }
       }
-    }
-    if (mutable_cache_) {
-      auto it = mutable_cache_->NewIterator(leveldb::ReadOptions());
-      it->Seek(key);
-      while (it->Valid()) {
-        auto cached_key = it->key().ToString();
-        if (cached_key.size() >= key.size() &&
-            std::equal(key.begin(), key.end(), cached_key.begin())) {
-          AddKeyLru(cached_key, it->value());
-        } else {
-          break;
+
+      if (mutable_cache_ && mutable_cache_lru_) {
+        auto it = mutable_cache_->NewIterator(leveldb::ReadOptions());
+        it->Seek(key);
+        while (it->Valid()) {
+          auto cached_key = it->key();
+          if (cached_key.size() >= key.size() &&
+              std::equal(key.begin(), key.end(), cached_key.data())) {
```

Comment:
Maybe directly use memcmp directly like leveldb::Slice is using?
See https://github.com/google/leveldb/blob/main/include/leveldb/slice.h#L95.